test/Microsoft.SqlTools.ServiceLayer.UnitTests/ObjectExplorer/NodeFilterTests.cs

```diff
@@ -610,5 +613,307 @@ public void TextNumericFilters()
             };
             Assert.Throws<IndexOutOfRangeException>(() => INodeFilter.GetPropertyFilter(filterList, typeof(SqlHistoryTableQuerier), ValidForFlag.All), "Error not thrown when the array contains single value for between operator");
         }
+
+        [Test]
```

Comment:
It would be good to add tests for special cases - such as strings with reserved characters (`'` and `%`)